features/step_definitions/cli_steps.js

```diff
@@ -6,145 +6,149 @@ import rimraf from 'rimraf'
+        if (err2) throw new Error(err2)
+        callback()
       })
     })
+  },
+)
 
-  this.Then(/^the exit status should be ([0-9]+)$/, (code, callback) => {
-    const actualCode = this.lastRun.error ? this.lastRun.error.code : '0'
-
-    if (`${actualCode}` !== `${code}`)
-      throw new Error(`Exit code expected: '${code}\n
-                      Got: '${actualCode}'\n
-                      ${getAdditionalErrorText(this.lastRun)}`)
-    callback()
-  })
-
-  this.Then(/^the output should contain:$/, (expectedOutput, callback) => {
-    const actualOutput = normalizeText(this.lastRun.error ?
-      this.lastRun.stderr : this.lastRun.stdout)
-    const normalizedExpectedOutput = normalizeText(expectedOutput)
 
-    if (actualOutput.indexOf(normalizedExpectedOutput) < 0)
-      throw new Error(`Expected output to match the following:\n
-                      '${normalizedExpectedOutput}'\n
-                      Got:\n'${actualOutput}'.\n
-                      ${getAdditionalErrorText(this.lastRun)}`)
-    callback()
-  })
-
-  this.Then(/^the output should be a version number$/, (callback) => {
-    const actualOutput = normalizeText(this.lastRun.stdout)
+When(/^I run `sphere(| .+)`$/, (args, callback) => {
+  const initialCwd = process.cwd()
+  process.chdir(tmpDir)
 
-    if (!new RegExp(/\d+\.\d+\.\d+/).test(actualOutput))
-      throw new Error(`Expected output to be a version number:\n
-                      Got:\n'${actualOutput}'.\n
-                      ${getAdditionalErrorText(this.lastRun)}`)
+  const runtimePath = joinPathSegments([ baseDir, 'bin', 'sphere' ])
+  const command = runtimePath + args
+  exec(command, (error, stdout, stderr) => {
+    lastRunSequence = { error, stdout, stderr }
+    process.chdir(initialCwd)
+    cleansingNeeded = true
     callback()
   })
-}
+})
+
+When(
+  /^I run with accessToken `sphere(| .+)`$/,
+  { timeout: 60 * 1000 },
+  (args, callback) => {
+    loadCredentials().then((credentials) => {
+      const oauth2 = new OAuth2({ config: credentials })
+      oauth2.getAccessToken((err, response, body) => {
+        if (err)
+          throw new Error(`Error occurred when fetching accessToken ${err}`)
+        const accessToken = body.access_token
+        const initialCwd = process.cwd()
+        process.chdir(tmpDir)
+
+        const runtimePath = joinPathSegments([ baseDir, 'bin', 'sphere' ])
+        const command = `${runtimePath} ${args} --accessToken ${accessToken}`
+        const env = Object.assign({}, process.env, {
+          SPHERE_CLIENT_ID: undefined,
+          SPHERE_CLIENT_SECRET: undefined,
+        })
+        const options = { env }
+        exec(command, options, (error, stdout, stderr) => {
+          lastRunSequence = { error, stdout, stderr }
+          process.chdir(initialCwd)
+          cleansingNeeded = true
+          callback()
+        })
+      })
+    })
+  },
+)
+
+Then(/^the exit status should be ([0-9]+)$/, (code, callback) => {
```

Comment:
I thought that cucumber should already know these "commands", why we define them here?

Reply:
For example here https://github.com/sphereio/sphere-product-type-json-generator/blob/master/features/common.feature#L5 we do not define what does "Then the exit status should be 1" mean and it works..

Reply:
Yeah. I think the more assertions the better. And as the test passes, I don't see a reason to remove it